app/controllers/api/v1/moves_controller.rb

```diff
@@ -73,6 +74,25 @@ def find_move
           .includes(:from_location, :to_location, person: { profiles: %i[gender ethnicity] })
           .find(params[:id])
       end
+
+      def import_moves_from_nomis
+        return unless from_location_nomis_agency_id && date
+
+        moves = NomisClient::Moves.get(from_location_nomis_agency_id, date)
+        Moves::Importer.new(moves).call
+      end
+
+      def from_location_nomis_agency_id
+        return unless filter_params[:from_location_id]
+
+        Location.find(filter_params[:from_location_id]).nomis_agency_id
```

Comment:
You could use `find_by(from_location_id: filter_params[:from_location_id])&.nomis_agency_id` here in case the supplied `from_location_id` doesn't resolve. Depends if you want to fail gracefully or throw an exception.

Reply:
I reckon that if a request specifies a `from_location_id` that doesn't exist we should return some kind of error. Maybe a `422` rather than a `500` though.

Reply:
in the `ApiController` what we do is

```
rescue_from ActiveRecord::RecordNotFound, with: :render_resource_not_found_error
```

I think that returning a 404 error is the most appropriate thing here, which is in line with what we do elsewhere, but happy to discuss about it.

Reply:
makes sense to me